tls/s2n_record_read.c

```diff
@@ -107,6 +107,9 @@ int s2n_record_header_parse(
                             != version,
             S2N_ERR_BAD_MESSAGE);
 
+    if (!conn->client_hello_record_version) {
+        conn->client_hello_record_version = version;
+    }
```

Comment:
Is it possible for the client hello record version to be 0 0? Then, this would only be updated on the next record, and wouldn't actually reflect the client hello record version.

Reply:
That is a really good question. It would certainly be a bug, and I'm not sure if we would succeed in the actual negotiation, but ... theoretically it is possible. I'll switch the if-statement to some other signal.